share/wake/lib/core/json.wake

```diff
@@ -169,3 +169,143 @@ export def x ==/ y = match x y
   (JArray a) _ = exists (_ ==/ y) a
   _ (JArray b) = exists (_ ==/ x) b
   _ _ = False
+
+# Simplify a JSON structure according to the rules given.
+# For example, Wake's implementation of JSON is more lenient than the base
+# standards, so if maximum compatibility is desired in situations where some
+# other implementation may be parsing generated output, `normalizeJSONCompat`
+# may be used to target that lowest common denominator.
+#
+# Note that the relevant normalization is applied to each member of a `JArray`
+# or `JObject` *before* the enclosing container. This means that if some
+# normalization fails anywhere in the JSON tree the entire call will fail, even
+# if the specific branch causing the failure would later have been pruned.
+export def normalizeJSONWith (fmt: JSONNormalize) (json: JValue): Result JValue Error =
```

Comment:
> Perhaps call this `normalizeJSONWith`, and then we can have a `normalizeJSON` which defaults to the json4 variant.

`With`, sure, and we can *maybe* add a plain alias later, but I think this is probably something that every project using wake would have different opinions on depending on their toolchain: some might want the most compatibility, some might only need canonical strings, some might require objects conform a stricter schema...

Reply:
Well, I totally agree we need to have an escape hatch to give people alternative normalizations. However, since we've decided JSON5 is non-standard, and IMO your 'compat' version makes quite sensible and non-controversial choices, I think it makes sense to stream-line the common case. People probably want to just do `normalizeJSON myJSON` and not worry about the tuple.